### ISSUE TITLE
Improper line numbers with #ifdef..#endif

### ISSUE THREAD

**Issue by theKidOfArcrania** [acowley/hpp#6]
Hi it appears that hpp produces some erratic line numbers. When hpp sees a `#ifdef`..`#endif` (without an `#else`), the line numbers will be different to what is actually in the original file. Here I have a file called `test` with the line numbers annotated at the beginning:
```c
#ifdef FOO
2: __LINE__: aaaa
#endif
4: __LINE__:
#ifdef FOO
6: __LINE__: aaaa
#endif
8: __LINE__:
#ifdef FOO
10: __LINE__: aaaa
#endif
```

When I run `hpp test` it produces:
```c
#line 4
4: 3:
#line 7
8: 6:
#line 10
```

And when I run `hpp -DFOO test` it produces:
```c
2: 2: aaaa
#line 4
4: 5:
6: 7: aaaa
#line 9
8: 10:
10: 12: aaaa
#line 14
```

**Comment by acowley** [comment]
Those look like good tests! I don’t have a fast answer for you, but this is a great start.

**Comment by hsyl20** [comment]
Fixed by #11
